src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -5111,22 +5112,17 @@ ProjectionNames QueryAnalyzer::resolveFunction(QueryTreeNodePtr & node, Identifi
     }
 
     /// Resolve function arguments
-
+    FunctionSecretArgumentsFinder::Result secret_arguments = FunctionSecretArgumentsFinder{function_node_ptr->toAST()->as<ASTFunction &>()}.getResult();
     bool allow_table_expressions = is_special_function_in;
     auto arguments_projection_names = resolveExpressionNodeList(function_node_ptr->getArgumentsNode(),
         scope,
         true /*allow_lambda_expression*/,
-        allow_table_expressions /*allow_table_expression*/);
+        allow_table_expressions /*allow_table_expression*/,
+        {secret_arguments.start, secret_arguments.count});
+
+    for (size_t n = secret_arguments.start; n < secret_arguments.start + secret_arguments.count; ++n)
+        arguments_projection_names[n] = "[HIDDEN]";
```

Comment:
Let's better call `setMasked` here. Just take `n`th argument, convert it to `ConstantNode` and mark as a secret.